### PR TITLE
Added websocketAsync and request to http and added websocket events to os

### PR DESCRIPTION
**Pull request by unlimitedcoder2**

### CODE REVIEW

**Review by Eforen — APPROVED** [review]
LGTM Thank you for the corrections

File: declarations/http.ts [hunk 1]
```diff
@@ -1,5 +1,47 @@
...
+         */
+        headers?: Headers;
+        /**
+         * Whether to make a binary HTTP request. If true, the body will not be UTF-8 encoded, and the received response will not be decoded.
+         */
+        binary?: boolean;
+        /**
+         * Which HTTP method to use, for instance "PATCH" or "DELETE".
+         */
+        method?: AnyCase<HttpMethod>;
```
Comment (Eforen) [inline declarations/http.ts:39]: Quick thought on this while I was reviewing it to publish to NPM, Is it truly case insensitive @unlimitedcoder2 ?

**Comment by Eforen** [comment]
This PR has been pushed up to NPM

File: declarations/http.ts (hunk 1 above)
Comment (unlimitedcoder2) [reply declarations/http.ts:39]: The methods are uppercased before they are converted to the java HttpMethod [code](https://github.com/SquidDev-CC/CC-Tweaked/blob/mc-1.15.x/src/main/java/dan200/computercraft/core/apis/HTTPAPI.java#L117)

Comment (Eforen) [reply declarations/http.ts:39]: Awsome! Nice catch!

**Pull request merged**